File: queue.js
class Queue {
  constructor(size) {
    this.array = new Array(size);
    this.length = 0;
    this.rear = 0;
    this.front = 0;
  }

  enqueue = (data) => {
    if (!data) {
      console.log("Invalid Argument");
      return;
    }
    if (this.length == this.array.length) {
      console.log("Queue is Full");
      return;
    }
    this.array[this.rear] = data;
    this.rear = (this.rear + 1) % 5; //circular queue implemention
    this.length++;
    console.log(this.array);
  };

  dequeue = () => {
    if (this.length == 0) {
      console.log("Queue is empty");
      return;
    }
    let x = this.array[this.front];
    delete this.array[this.front];
    this.front = (this.front + 1) % 5; //circular queue implemention
    this.length--;
    console.log(x);
    console.log(this.array);
    return x;
  };
}

let queue = new Queue(5);

queue.enqueue(2);
queue.enqueue(5);
queue.enqueue(10);
queue.enqueue(20);
queue.enqueue(30);
queue.dequeue();
queue.dequeue();
queue.enqueue(40);

File: queue.py
class Queue:
    def __init__(self, size):
        self.front = 0
        self.rear = 0
        self.length = 0
        self.array = [None] * size
    
    def enqueue(self, data = None):
        if data == None:
            print("Argument shouldn't be empty")
            return
        if self.length == len(self.array):
            print("Queue is full")
            return
        self.array[self.rear] =  data
        self.rear = (self.rear + 1) % 5 #circular queue implemention
        self.length += 1
        print(self.array)

    def dequeue(self):
        if self.length == 0:
            print("Queue is empty")
            return
        x = self.array[self.front]
        self.array[self.front] = None
        self.front = (self.front + 1) % 5 #circular queue implemention
        self.length -= 1
        print(x)
        print(self.array)


queue = Queue(5)

queue.enqueue(1)
queue.enqueue(2)
queue.enqueue(3)
queue.enqueue(4)
queue.dequeue()
queue.dequeue()
queue.enqueue(5)
queue.enqueue(6)